monix-execution/jvm/src/main/scala/monix/execution/misc/Local.scala

```diff
@@ -0,0 +1,104 @@
+package monix.execution.misc
+
+import java.util.UUID
+
+object Local {
+
+  type Context = scala.collection.immutable.Map[String, Option[LocalContext]]
+
+  trait LocalContext
+
+  private[this] def context(k: String, v: Option[LocalContext]): Context =
+    scala.collection.immutable.Map(k -> v)
+
+  private[this] val localContext = ThreadLocal[Context]()
+
+  private def register(): String = synchronized {
+    UUID.randomUUID().toString
+  }
+
+  /**
+    * Return the state of the current Local state.
```

Comment:
So `Local` needs more ScalaDoc. I suggest also coming up with some dumb samples for these functions.

Reply:
@alexandru Understood. The samples on the tests or in the comments?